src/notebook/Cell.jl

```diff
@@ -64,3 +66,16 @@ end
 function Base.convert(::Type{UUID}, string::String)
     UUID(string)
 end
+
+function get_cell_metadata(cell::Cell)::Dict{String,Any}
+    Dict(
+        if cell.running_disabled
```

Comment:
The first thing I can definitively do.

Regarding removing `depends_on_disabled_cells `and `running_disabled` - this may cause a performance impact because this information is used on every cell evaluation, and the metadata dictionary is not concretely typed.
No idea if this makes a significant difference, however.

Reply:
My idea was that we could just give a single metadata API to Javascript/Julia, and cell disabling would be handling directly as a result but we can still change this later.

Reply:
Sounds good.

The metadata is now published to the frontend in `cell_input.metadata`. Changing the cell disabling from using separate fields to metadata itself has quite a large impact, therefore I would suggest to do it later (if required at all).